auth_service/auth_adapter/core/auth.py

```diff
@@ -20,15 +20,51 @@
 import logging
 from typing import Any, Optional, TypedDict
 
-from jwcrypto import jws
+from jwcrypto import jwk, jws
 
-from .jwks import external_jwks, internal_jwk
+from ...config import Config
 
-__all__ = ["exchange_token"]
+__all__ = ["exchange_token", "signing_keys"]
 
 log = logging.getLogger(__name__)
 
 
+class SigningKeys:
+    """A container for external and internal signing keys."""
+
+    external_jwks: jwk.JWKSet  # the external public key set
+    internal_jwk: jwk.JWK  # the interal key pair
+
+    # when testing, we also provide the extrnal private key
+    full_external_jwk: Optional[jwk.JWK] = None
+
+    def load(self, config: Config) -> None:
+        """Load all the signing keys from the configuration."""
+        external_keys = config.auth_ext_keys
+        if not external_keys:
```

Comment:
What is the advantage of having this here? Maybe we can consider to generate test credentials as part of a pytest fixture.

Reply:
Will be refactored as discussed.